src/librustdoc/html/format.rs

```diff
@@ -822,9 +822,23 @@ pub(crate) fn print_anchor(did: DefId, text: Symbol, cx: &Context<'_>) -> impl D
     fmt::from_fn(move |f| {
         let parts = href(did, cx);
         if let Ok((url, short_ty, fqp)) = parts {
+            let tcx = cx.tcx();
+            let def_kind = tcx.def_kind(did);
+            let anchor = if matches!(
+                def_kind,
+                DefKind::AssocTy | DefKind::AssocFn | DefKind::AssocConst | DefKind::Variant
+            ) {
+                let parent_def_id = tcx.parent(did);
+                let item_type =
+                    ItemType::from_def_kind(def_kind, Some(tcx.def_kind(parent_def_id)));
+                format!("#{}.{}", item_type.as_str(), tcx.item_name(did))
+            } else {
+                String::new()
+            };
```

Comment:
This duplicates code already present in `assoc_href_attr`.  Please either call that function, or factor out their common code into a shared function.

Also, I think the functions `href`, `make_href`, and `href_with_root_path` should have doc comments explaining that they do not include the fragment.

Reply:
I opened https://github.com/rust-lang/rust/issues/148547 with an explanation of what I have in mind (and what I already tried too 😉 ).